book/src/ephemeral-local-volumes.md

```diff
@@ -1,38 +1,49 @@
-# Ephemeral Local Volumes
+# Pod Inline Volume Support
 
-> ## *This page is still under active development.*
+## Status
 
-**Status:** Alpha
+Status | Min K8s Version | Max K8s Version 
+--|--|--
+Alpha | 1.15 | - 
 
-Kubernetes supports three types of volumes:
+## Overview
+Traditionally, volumes that are backed by CSI drivers can only be used with a `PersistentVolume` and `PersistentVolumeClaim` object combination. This feature supports ephemeral storage use cases and allows CSI volumes to be specified directly in the pod specification.  At runtime, nested inline volumes follow the ephemeral lifecycle of their associated pods where the driver handles all phases of volume operations as pods are created and destroyed.
 
-1. Remote Persistent Volumes
-2. Local Persistent Volumes
-3. Local Ephemeral Volumes
+> See the [design document](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190122-csi-inline-volumes.md) for futher information.
 
-The initial focus of Kubernetes CSI was Remote Persistent Volumes. However, the goal is for CSI to support all three types.
 
-This page documents how to create "Local Ephemeral Volumes" for Kubernetes using CSI.
+### Example of inline CSI pod spec
+A pod spec with an ephemeral inline CSI volume. Note that because the volume is expected to be ephemeral, the volumeHandle is not provided. Instead a CSI-generated ID will be submitted to the driver.
 
-# What is a Local Ephemeral Volumes?
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: some-pod
+spec:
+  containers:
+    ...
+  volumes:
+      - name: vol
+        csi:
+          driver: inline.storage.kubernetes.io
+          volumeAttributes:
+              foo: bar
```

Comment:
Not necessarily trying to promote the image driver, but the example from there may be more concrete an example and more easy to understand? This one is a little generic. Should we use that example?

```yaml
apiVersion: v1
kind: Pod
spec:
  containers:
  - image: nginx:1.13-alpine
    volumeMount:
    - name: data
      mountPath: /usr/share/nginx/html
  volumes:
  - name: data
    csi:
      driver: image.csi.k8s.io
      volumeAttributes:
          image: kfox1111/misc:test
```

Reply:
I think that makes sense. But then we should also link to the documentation for the driver, either in a comment above `driver: ...` or in the introduction of the example. The latter would have the advantage that it can be a clickable link.

Reply:
My only concern about this example is the reference to a private, unknown image. It may be better to refer to some publicly known image like busybox

Reply:
Seems like the consensus here it to keep the example as is using the Hostpath.

Reply:
The only weird there is it shows an example where a regular container (executable binaries) is being shared out by a webserver. It may be confusing to the reader. "why would you put busybox there?"

Should we produce a container that could be used for this purpose? Something like:
kubernetes-csi/test-ephemeral-image

index.html:
Hello World!

Dockerfile:
FROM scratch
COPY index.html /index.html

?

Reply:
> Seems like the consensus here it to keep the example as is using the Hostpath.

As is works for me.

HostPath is kind of confusing, as it raises the question, why not just emptyDir?

Reply:
@kfox1111 please raise this point against the hostpath driver itself so that this documentation PR can keep moving.

Reply:
The example doesn't really help someone who's new to the topic. For example, "inline.storage.kubernetes.io" looks like some magic string that always has to be there. It's not obvious that this the name of a fake CSI driver. Same with "foo" and "bar" - what is that about?

How about this:
```
volumes:
  - name: data
    csi:
      driver: some-csi-driver.example.com
      # volumeAttributes contains arbitrary key/value pairs,
      # valid options depend on the driver.
      volumeAttributes:
          dataSource: from-somewhere
```